src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -192,15 +204,11 @@ public E borrowItemWithTimeout(long timeout, TimeUnit unit) throws
 
     @Override
     public void releaseItem(E e) {
-        releaseItem(e, true);
-    }
-
-    @Override
-    public void releaseItem(E e, boolean returnToPool) {
         final ReentrantLock lock = this.queueLock;
         lock.lock();
         try {
-            if (returnToPool) {
+            // Do nothing if the item being returned is the poisonPill
```

Comment:
It probably should be, but right now it isn't impossible. Because right now, it would be possible to return a different poison pill if:
1. You insert a pill
2. Borrow an instance (the pill)
3. Insert a different pill, overwriting the first
4. Return the first pill

We'll have to decide how should we handle two successive calls to `.insertPill`